File: Python/Arrays/Easy/TwoSum.py
# Intuition
'''
'''

# Approach
'''
'''

# Complexity

# - Time complexity:
'''
O(N) as we need to traverse the given list to populate our dictionary (Beats 100% of solutions on LeetCode)
'''

# - Space complexity:

'''
'''

from typing import List

def twoSum(self, nums: List[int], target: int) -> List[int]:
    d = {}
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement in d:
            return [i, d[complement]]
        d[nums[i]] = i

File: Cpp/Arrays/Easy/TwoSum.cpp
/*
Intuition


Approach


Complexity
- Time complexity:

- Space complexity:

*/

#include <vector>
#include <unordered_map>

std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> map;

    for(int i = 0; i < nums.size(); i++){
        int complement = target - nums[i];
        if(map.find(complement) != map.end()){
            return {map[complement], i};
        }
        map[nums[i]] = i; 
    }
    return {-1, -1};
}